.github/scripts/qemu_test.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+cd build/exe/cpu1
+for i in *testrunner.exe *UT.exe *test.exe; do \
+  file=log-${i%%.exe}.txt
+  rm $file
+  qemu-system-i386 -m 128 -kernel $i -nographic -no-reboot -append "--console=/dev/com1 --video=off" \
+       | while read line
+  do
+    echo $line | tee -a $file;
+    echo $line | grep "Init thread idle" | while read line2
+      do
+        pkill qemu-system- 
+      done
+  done
```

Comment:
Here is the command template I typically use for running QEMU+RTEMS 4.11.  It's from a makefile rule, so $(<) represents the "exe" name.  The ".cow" is a temporary copy-on-write image that provides a writable filesystem (some tests e.g. module loader need this).  Serial output is first written to a tmp file, then renamed to a log file that can be captured/preserved.  Importantly, the `--batch-mode` append option causes QEMU to self-exit after the test completes, which removes the need to monitor the stdout like that.

        $(QEMU_COMMAND) -no-reboot -display none \
            -kernel $(<) \
            -append '--batch-mode' \
            -drive file=$(*).cow,format=qcow2 \
            -device i82557b,netdev=net0,mac=$(MACADDR) \
            -netdev user,id=net0 \
            -serial file:$(@).tmp
        @mv -v $(@).tmp $(@)

Reply:
:+1: Resolved